tests/decode_string_test.go

```diff
@@ -0,0 +1,51 @@
+package tests
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/paribu/event-decoder/event"
+)
+
+func TestDecodeString(t *testing.T) {
+	abiFile := "decode_string_abi.json"
+
+	encodedEvent := &event.Event{
+		Topics: []common.Hash{
+			common.HexToHash("0x402ddd519dec94a9c869d15434d5df9d0be41d3d937c9dbe2157489dc9a2be0d"),
+		},
+		Data: "0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002a30786135303864643837356631306333336335326138616262323065313666633638653938316631383600000000000000000000000000000000000000000000",
+	}
+
+	expectedParameters := []event.Parameter{
+		{
+			Type:  "string",
+			Name:  "newValue",
+			Value: "0xa508dD875f10C33C52a8abb20E16fc68E981F186",
+		},
+	}
+
+	parameters, err := decodeEventWithABI(abiFile, encodedEvent)
+	if err != nil {
+		t.Error(err)
+	}
+
+	for i, decoded := range parameters {
+		expected := expectedParameters[i]
+
+		if expected.Name != decoded.Name {
+			t.Errorf("expected name %s, got %s", expected.Name, decoded.Name)
+		}
+		if expected.Type != decoded.Type {
+			t.Errorf("expected type %s, got %s", expected.Type, decoded.Type)
+		}
+
+		expectedValueTrim := strings.Trim(strings.ToLower(expected.Value), "")
```

Comment:
You don't need "Trim" functions here and one line below, because it already gets an empty string as a parameter. It does nothing.

You also don't need to use "ToLower" function. The error probably happens because received / decoded value does not contains checksum. Instead of ToLower, do this:

decoded.Value = common.HexToAddress(decoded.Value).Hex()

After that you can compare them.